shared/src/price_estimate.rs

```diff
@@ -98,6 +114,10 @@ impl PriceEstimating for UniswapPriceEstimator {
         amount: U256,
         kind: OrderKind,
     ) -> Result<BigRational> {
+        match self.token_validation(&buy_token, &sell_token) {
+            Ok(_) => (),
+            Err(validation_err) => return Err(validation_err),
+        }
```

Comment:
I am not terribly impressed with the lack of savings obtained with the `token_validation` method given that I needed a match statement to parse this. I am sure there is a cleaner way to express the same logic here, but I am not sure what it is.